langlib/lang.float/src/main/java/org/ballerinalang/langlib/floatingpoint/ToExpString.java

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.langlib.floatingpoint;
+
+import io.ballerina.runtime.api.creators.ErrorCreator;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.values.BString;
+import io.ballerina.runtime.internal.FloatUtils;
+import io.ballerina.runtime.internal.util.exceptions.BLangExceptionHelper;
+import io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons;
+import io.ballerina.runtime.internal.util.exceptions.RuntimeErrors;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.FLOAT_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Native implementation of lang.float:toExpString(float, int?).
+ *
+ * @since 2201.1.0
+ */
+public class ToExpString {
+
+    public static BString toExpString(double x, Object fractionDigits) {
+        // If `x` is NaN or infinite, the result will be the same as `value:toString`.
+        BString str = FloatUtils.getBStringValue(x);
+        if (str != null) {
+            return StringUtils.fromString(StringUtils.getStringValue(x, null));
+        }
+
+        long noOfFractionDigits;
+
+        // If fractionalDigits is `()`, use the minimum number of digits required to accurately represent the value.
+        double xAbsValue = Math.abs(x);
+        if (fractionDigits == null) {
+            if (xAbsValue == 0) {
+                noOfFractionDigits = BigDecimal.valueOf(xAbsValue).scale();
```

Comment:
Wouldn't `BigDecimal.valueOf(xAbsValue).scale()` for 0 will always be 1

Reply:
yes. Since the value is a `float` that's okay right?

Reply:
My point is; we can directly assign 1 instead of the function call

Reply:
ah yes! will change.